openupgrade_scripts/scripts/mass_mailing/16.0.2.5/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,113 @@
+---Models in module 'mass_mailing'---
+new model mailing.contact.import [transient]
+new model mailing.filter
+#NOTHING TO DO: new model
+
+---Fields in module 'mass_mailing'---
+mass_mailing / ir.mail_server           / active_mailing_ids (one2many) : NEW relation: mailing.mailing
+mass_mailing / mailing.contact.subscription / _order                        : _order is now 'list_id DESC, contact_id DESC' ('id')
+mass_mailing / mailing.filter           / mailing_domain (char)         : NEW required
+mass_mailing / mailing.filter           / mailing_model_id (many2one)   : NEW relation: ir.model, required
+mass_mailing / mailing.filter           / name (char)                   : NEW required
+mass_mailing / mailing.list             / _order                        : _order is now 'create_date DESC' ('name')
+mass_mailing / mailing.mailing          / _inherits                     : DEL _inherits: {'utm.source': 'source_id'}
+mass_mailing / mailing.mailing          / _order                        : _order is now 'calendar_date DESC' ('sent_date DESC')
+mass_mailing / mailing.mailing          / copyvalue (char)              : DEL
+mass_mailing / mailing.mailing          / favorite (boolean)            : NEW
+mass_mailing / mailing.mailing          / favorite_date (datetime)      : NEW isfunction: function, stored
+mass_mailing / mailing.mailing          / mailing_filter_id (many2one)  : NEW relation: mailing.filter, hasdefault: compute
+mass_mailing / mailing.mailing          / model_object_field (many2one) : DEL relation: ir.model.fields
+mass_mailing / mailing.mailing          / null_value (char)             : DEL
+mass_mailing / mailing.mailing          / sub_model_object_field (many2one): DEL relation: ir.model.fields
+mass_mailing / mailing.mailing          / sub_object (many2one)         : DEL relation: ir.model
+# NOTHING TO DO
```

Comment:
I think you should create mailing.filter records for each sub_object in mailing.mailing and then fill mailing_filter_id.

Reply:
but sub_object is not store, I don't know how to create mail_filter from it

Reply:
> but sub_object is not store, I don't know how to create mail_filter from it

I think you can extract sub_object data from ir.model and ir.model.field

Reply:
`mailing.filter` are like search favorites, but inside the mailing, so no need to perform anything in this table. It's a new feature.